src/lib.rs

```diff
@@ -54,6 +54,19 @@ fn ensure_current_version<'de, D: Deserializer<'de>>(d: D) -> Result<String, D::
     Ok(unchecked)
 }
 
+#[derive(Clone, Serialize, Deserialize, Debug, PartialEq, Eq, Default)]
+pub struct BuildInfo {
+    /// The hash of the contract WASM code.
+    pub wasm_hash: String,
```

Comment:
Opinionated: This doesn't seem related to the build info at all, it's just the code hash. Wondering if it should just be in the root metadata?

Reply:
Hmm, fair, especially because it also resolved Mira's point above